ui/src/app/machines/views/MachineDetails/MachineStorage/AvailableStorageTable/AvailableStorageTable.test.tsx

```diff
@@ -73,11 +73,85 @@ describe("AvailableStorageTable", () => {
     );
 
     expect(wrapper.find("tbody TableRow").length).toBe(1);
-    expect(wrapper.find("TableCell DoubleRow").at(0).prop("primary")).toBe(
+    expect(wrapper.find("TableCell DoubleRow").at(0).find("label").text()).toBe(
       availableDisk.name
     );
   });
 
+  it("can select a single disk", () => {
+    const disk = diskFactory({
+      available_size: MIN_PARTITION_SIZE + 1,
+      filesystem: null,
+      type: "physical",
+    });
+    const state = rootStateFactory({
+      machine: machineStateFactory({
+        items: [
+          machineDetailsFactory({
+            disks: [disk],
+            system_id: "abc123",
+          }),
+        ],
+      }),
+    });
+    const store = mockStore(state);
+    const wrapper = mount(
+      <Provider store={store}>
+        <AvailableStorageTable canEditStorage systemId="abc123" />
+      </Provider>
+    );
+
+    wrapper
+      .find("TableCell input")
+      .at(0)
+      .simulate("change", {
+        target: { name: disk.id },
+      });
+
+    expect(wrapper.find("TableCell Input").prop("checked")).toBe(true);
+  });
+
+  it("can select all disks", () => {
+    const disks = [
+      diskFactory({
+        available_size: MIN_PARTITION_SIZE + 1,
+        filesystem: null,
+        type: "physical",
+      }),
+      diskFactory({
+        available_size: MIN_PARTITION_SIZE + 1,
+        filesystem: null,
+        type: "physical",
+      }),
+    ];
+    const state = rootStateFactory({
+      machine: machineStateFactory({
+        items: [
+          machineDetailsFactory({
+            disks: disks,
+            system_id: "abc123",
+          }),
+        ],
+      }),
+    });
+    const store = mockStore(state);
+    const wrapper = mount(
+      <Provider store={store}>
+        <AvailableStorageTable canEditStorage systemId="abc123" />
+      </Provider>
+    );
+
+    wrapper
+      .find("TableHeader input")
+      .at(0)
+      .simulate("change", {
+        target: { name: "all-disks-checkbox" },
+      });
+
+    expect(wrapper.find("TableCell Input").at(0).prop("checked")).toBe(true);
+    expect(wrapper.find("TableCell Input").at(1).prop("checked")).toBe(true);
```

Comment:
There are [every](https://enzymejs.github.io/enzyme/docs/api/ReactWrapper/every.html) and [everyWhere](https://enzymejs.github.io/enzyme/docs/api/ReactWrapper/everyWhere.html) assertions you could use here if you felt like it.

Reply:
Oh nice, I didn't know about them. Defs file like this is a good use for them